csharp/src/Ice/Endpoint.cs

```diff
@@ -2,202 +2,137 @@
-
-        public abstract string Options();
+        // Returns a new endpoint with a different compression flag, provided that compression is supported by the
+        // endpoint. Otherwise the same endpoint is returned.
+        public abstract Endpoint NewCompressionFlag(bool compressionFlag);
 
-        public virtual void InitWithOptions(List<string> args)
-        {
-            var unknown = new List<string>();
+         // Returns an acceptor for this endpoint, or null if no acceptor is available.
+        public abstract IAcceptor? Acceptor(string adapterName);
 
-            string str = "`" + Transport() + " ";
-            foreach (string p in args)
-            {
-                if (IceUtilInternal.StringUtil.FindFirstOf(p, " \t\n\r") != -1)
-                {
-                    str += " \"" + p + "\"";
-                }
-                else
-                {
-                    str += " " + p;
-                }
-            }
-            str += "'";
+        // Returns a connector for this endpoint, or empty list if no connector is available.
+        public abstract void ConnectorsAsync(EndpointSelectionType endpointSelection, IEndpointConnectors callback);
 
-            for (int n = 0; n < args.Count; ++n)
-            {
-                string option = args[n];
-                if (option.Length < 2 || option[0] != '-')
-                {
-                    unknown.Add(option);
-                    continue;
-                }
-
-                string? argument = null;
-                if (n + 1 < args.Count && args[n + 1][0] != '-')
-                {
-                    argument = args[++n];
-                }
-
-                if (!CheckOption(option, argument, str))
-                {
-                    unknown.Add(option);
-                    if (argument != null)
-                    {
-                        unknown.Add(argument);
-                    }
-                }
-            }
+        // Expands endpoint out in to separate endpoints for each local host if listening on INADDR_ANY on server side
+        // or if no host was specified on client side.
+        public abstract List<Endpoint> ExpandIfWildcard();
 
-            args.Clear();
-            args.AddRange(unknown);
-        }
+        // Expands endpoint out into separate endpoints for each IP address returned by the DNS resolver. Also returns
+        // the endpoint which can be used to connect to the returned endpoints or null if no specific endpoint can be
+        // used to connect to these endpoints (e.g.: with the IP endpoint, it returns this endpoint if it uses a fixed
+        // port, null otherwise).
+        public abstract List<Endpoint> ExpandHost(out Endpoint? publishedEndpoint);
 
-        protected virtual bool CheckOption(string option, string? argument, string endpoint) =>
-            // Must be overridden to check for options.
-            false;
+         public abstract ITransceiver? GetTransceiver();
```

Comment:
Why the rename here? If we rename Transceiver to GetTransceiver shouldn't we also rename Acceptor/Connectors methods?

Reply:
`Transceiver` like `Endpoint` is/was a parameter-less method and as a result a candidate for becoming a get-only property, except it does not have property semantics (it returns null or a new object each time). So yes, I will add get to the  Acceptor and Connectors methods for consistency (these methods take parameters so are not candidates to become get properties).

Reply:
Renamed `Acceptor` to `GetAcceptor`. Did not rename `ConnectorsAsync` since it returns void.